aws/resource_aws_route_table_test.go

```diff
@@ -630,98 +654,721 @@ func TestAccAWSRouteTable_ConditionalCidrBlock(t *testing.T) {
+			}
+			if resp == nil || len(resp.RouteTables) == 0 {
+				return resource.NonRetryableError(fmt.Errorf("Route Table not found"))
+			}
+
+			for _, route := range resp.RouteTables[0].Routes {
+				if aws.StringValue(route.DestinationPrefixListId) == plId {
+					return nil
+				}
+			}
+
+			return resource.RetryableError(fmt.Errorf("Route not found"))
+		})
+
+		return err
+	}
+}
+
+func testAccAWSRouteTableConfigBasic(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_vpc" "test" {
+  cidr_block = "10.1.0.0/16"
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_route_table" "test" {
+  vpc_id = aws_vpc.test.id
+}
+`, rName)
+}
+
+func testAccAWSRouteTableConfigSubnetAssociation(rName string) string {
+	return composeConfig(testAccAvailableAZsNoOptInDefaultExcludeConfig(), fmt.Sprintf(`
+resource "aws_vpc" "test" {
+  cidr_block = "10.1.0.0/16"
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_subnet" "test" {
+  cidr_block        = "10.1.1.0/24"
+  vpc_id            = aws_vpc.test.id
+  availability_zone = data.aws_availability_zones.available.names[0]
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_route_table" "test" {
+  vpc_id = aws_vpc.test.id
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_route_table_association" "test" {
+  route_table_id = aws_route_table.test.id
+  subnet_id      = aws_subnet.test.id
+}
+`, rName))
+}
+
+func testAccAWSRouteTableConfigIpv4InternetGateway(rName, destinationCidr1, destinationCidr2 string) string {
+	return fmt.Sprintf(`
+resource "aws_vpc" "test" {
+  cidr_block = "10.1.0.0/16"
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_internet_gateway" "test" {
+  vpc_id = aws_vpc.test.id
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_route_table" "test" {
+  vpc_id = aws_vpc.test.id
+
+  route {
+    cidr_block = %[2]q
+    gateway_id = aws_internet_gateway.test.id
+  }
+
+  route {
+    cidr_block = %[3]q
+    gateway_id = aws_internet_gateway.test.id
+  }
+
+  tags = {
+    Name = %[1]q
+  }
+}
+`, rName, destinationCidr1, destinationCidr2)
+}
+
+func testAccAWSRouteTableConfigIpv6EgressOnlyInternetGateway(rName, destinationCidr string) string {
+	return fmt.Sprintf(`
+resource "aws_vpc" "test" {
+  cidr_block                       = "10.1.0.0/16"
+  assign_generated_ipv6_cidr_block = true
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_egress_only_internet_gateway" "test" {
+  vpc_id = aws_vpc.test.id
+
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_route_table" "test" {
+  vpc_id = aws_vpc.test.id
+
+  route {
+    ipv6_cidr_block        = %[2]q
+    egress_only_gateway_id = aws_egress_only_internet_gateway.test.id
+  }
+
+  tags = {
+    Name = %[1]q
+  }
+}
+`, rName, destinationCidr)
+}
+
+func testAccAWSRouteTableConfigIpv4Instance(rName, destinationCidr string) string {
+	return composeConfig(
+		testAccLatestAmazonNatInstanceAmiConfig(),
+		testAccAvailableAZsNoOptInDefaultExcludeConfig(),
```

Comment:
Nit: Shouldn't be anything in this test that needs exclusion of certain Availability Zones:

```suggestion
		testAccAvailableAZsNoOptInConfig(),
```